meerk40t/core/elements/elements.py

```diff
@@ -1345,6 +1360,110 @@ def create_image(oplist):
         if needs_signal:
             self.signal("default_operations")
 
+    def create_usable_copy(self, sourceop):
+        from copy import copy
+
+        op_to_use = copy(sourceop)
+        for attr in ("id", "label", "color", "lock", "allowed_attributes"):
+            setattr(op_to_use, attr, getattr(sourceop, attr))
+        return op_to_use
+
+    def set_default_id(self, targetop):
+        id_candidate = 0
+        pattern = "D"  # Default
+        op_name_parts = targetop.type.split(" ")
+        if len(op_name_parts) > 1:
+            pattern = op_name_parts[1].upper()[0]
+        id_existing = True
+        while id_existing:
+            id_existing = False
+            id_candidate += 1
+            op_id = f"{pattern}{id_candidate}"
+            for op in list(self.ops()):
+                if op_id == op.id:
+                    id_existing = True
+                    break
+        targetop.id = op_id
```

Comment:
Slight observation, not really actionable. But, `elements.validate_ids()` exists. It's typically though the use might not include default ops though `targetops` there might. There's some overlap there. `validate_ids" typically does `meerk40t:{idx}` in those cases when the id is missing or is duplicated and thus must be cleared. This is typically for saving an svg file and allowing valid references. But, there's not really much of an issue combining these schemes and calling the same code line when we're saving an SVG as we use other places to ensure our nodes have IDs when they are used for something.

Reply:
Extended and used the standard method